driver-core/src/main/java/com/datastax/driver/core/AbstractSession.java

```diff
@@ -51,6 +51,7 @@ public ResultSet execute(String query, Object... values) {
      */
     @Override
     public ResultSet execute(Statement statement) {
+        checkNotInEventLoop();
```

Comment:
Mmm actually I don't like that solution. `SessionManager` needs to override only some of the overloaded variants, to avoid performing the check twice (for example it's needed for `execute(Statement)` but not `execute(String)` because it calls the former). And that info is an implementation detail of `AbstractSession`, so I prefer keeping it there.
Implementations that don't need the check don't need to do anything, since by default the method is empty.